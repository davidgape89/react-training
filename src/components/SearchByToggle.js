import React from 'react';
import PropTypes from 'prop-types';

const SearchByToggle = ({onChange, value}) => {

  // Assign the color to the button according to whether it is
  // active or not
  const getClassList = (criteria) => {
    const color = value === criteria? 'button--red': 'button--grey';
    return `button button--small ${color}`;
  }

  return (
    <div className="search-by-toggle">
      <span>SEARCH BY:</span>
      <button className={getClassList('title')}
              onClick={() => onChange('title')}>
        TITLE
      </button>
      <button className={getClassList('genres')}
              onClick={() => onChange('genres')}>
        GENRE
      </button>
    </div>
  )
};

SearchByToggle.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func
};

export default SearchByToggle;