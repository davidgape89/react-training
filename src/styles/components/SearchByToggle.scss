@import '../styles/variables';

.search-by-toggle {
  align-items: center;
  display: flex;
  color: $white;
  font-size: 12px;
  font-weight: 600;
  line-height: 12px;

  > * {
    margin-right: $medium-size;
  }
}
