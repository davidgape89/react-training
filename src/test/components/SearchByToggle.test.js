import React from 'react';
import {shallow} from 'enzyme';

import SearchByToggle from '../../components/SearchByToggle';

describe('SearchByToggle -', () => {
  let wrapper, callbackFun;

  beforeEach(() => {
    callbackFun = jest.fn();
    wrapper = shallow(<SearchByToggle onChange={callbackFun}
                                      value="genre" />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('changes the selected button correctly', () => {
    wrapper.setProps({value: 'title'});
    const titleButton = wrapper.find('button').first();

    expect(titleButton.hasClass('button--red')).toBe(true);
  });

  it('calls the callback on title correctly', () => {
    wrapper.find('button')
           .first()
           .simulate('click');

    expect(callbackFun).toHaveBeenCalled();
  });

  it('calls the callback on genre correctly', () => {
    wrapper.find('button')
           .at(1)
           .simulate('click');

    expect(callbackFun).toHaveBeenCalled();
  });
});