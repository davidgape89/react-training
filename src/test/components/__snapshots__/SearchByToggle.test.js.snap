// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchByToggle - renders correctly 1`] = `
<div
  className="search-by-toggle"
>
  <span>
    SEARCH BY:
  </span>
  <button
    className="button button--small button--grey"
    onClick={[Function]}
  >
    TITLE
  </button>
  <button
    className="button button--small button--red"
    onClick={[Function]}
  >
    GENRE
  </button>
</div>
`;
